src/main/java/org/checkerframework/specimin/JavaParserUtil.java

```diff
@@ -222,14 +210,31 @@ public static String getValueTypeFromAnnotationExpression(Expression value) {
    * @param node a node that is contained in a class-like structure
    * @return the nearest enclosing class-like node
    */
-  public static Node getEnclosingClassLike(Node node) {
+  public static TypeDeclaration<?> getEnclosingClassLike(Node node) {
     Node parent = node.getParentNode().orElseThrow();
-    while (!(parent instanceof ClassOrInterfaceDeclaration
-        || parent instanceof EnumDeclaration
-        || parent instanceof AnnotationDeclaration)) {
+    while (!(parent instanceof TypeDeclaration<?>)) {
       parent = parent.getParentNode().orElseThrow();
     }
-    return parent;
+    return (TypeDeclaration<?>) parent;
+  }
+
+  /**
+   * See {@code getEnclosingClassLike(Node)} for more details. This does not throw if a parent is
+   * not found; instead, it returns null.
+   *
+   * @param node a node that is contained in a class-like structure
+   * @return the nearest enclosing class-like node
+   */
+  public static @Nullable TypeDeclaration<?> getEnclosingClassLikeOptional(Node node) {
+    Optional<Node> parent = node.getParentNode();
```

Comment:
Rather than two copies of code that logically does the same thing, could you implement this version in terms of the other (e.g., by catching an exception) or vice-versa? I think it should be relatively easy to do so.